### PR TITLE
Support array sequence format

### PR DESCRIPTION
**Pull request by jaydoane**
Allow test to pass with JSON sequence format like:
"seq": [num, binary]

### CODE REVIEW

File: test/chttpd_db_test.erl [hunk 1]
```diff
@@ -83,17 +83,19 @@ should_accept_live_as_an_alias_for_continuous(Url) ->
...
         {[{_, Seq}, _]} = ?JSON_DECODE(CleanedResult),
-        <<SeqNum0:1/binary, "-", _/binary>> = Seq,
-        SeqNum = list_to_integer(binary_to_list(SeqNum0)),
+        SeqNum = seq_num(Seq),
 
         ?assertEqual(LastSeqNum + 1, SeqNum)
     end).
+
+seq_num(<<SeqNumBin:1/binary, "-", _/binary>>) ->
+    list_to_integer(binary_to_list(SeqNumBin));
```
Comment (kxepal) [inline test/chttpd_db_test.erl]: it's possible now to use `binary_to_integer` as our minimal supported Erlang version is R16B03-1 which has it.

**Comment by jaydoane** [comment]
@kxepal, thanks for the tip. If you are inclined to accept this PR, I would be happy to open a Jira ticket for it.

**Comment by kxepal** [comment]
I think I do (after squash), no need for JIRA for such trivial changes. 
The only question is when array sequence format happens? And why we're not stable on seq format?

**Comment by jaydoane** [comment]
The only place I know that an array is used for seq format is in a soon-to-be legacy version of Cloudant Local. @mikewallace1979 probably has the backstory on where it originally came from.

I can understand any reluctance to adding something like this to couchdb, but the alternatives for keeping downstream testing green appear rather distasteful. My plan was to open this PR to get Mike's opinion on whether we should even pursue the idea of getting this merged upstream, but kxepal beat me to it.

**Comment by mikewallace1979** [comment]
@kxepal @jaydoane Yes we standardized on the string format in https://github.com/apache/couchdb-fabric/commit/42a67d0a018868d3760df90eca9d2e8e10614fd7 and the array format is only required for a downstream version which is nothing to do with the ASF.

That being the case I don't think we should merge this here.

**Comment by kxepal** [comment]
@mikewallace1979 I see. Agree, there is no point in this support then. I think there is nothing wrong for downstream to apply this patch and handle further changes via simple merge till this wouldn't be required.

Otherwise we'll end with a lot odd-and-legacy branches that nobody knows for what they have to be used.

**Comment by jaydoane** [comment]
Makes sense to me! Closing.

**Pull request closed without merging**